src/bin/help.ts

```diff
@@ -0,0 +1,172 @@
+import chalk from "chalk";
+
+import { allArgOptions } from "../shared/options/args.js";
+
+interface HelpTextSection {
+	sectionHeading: string;
+	subsections: {
+		flags: Flag[];
+		subheading: string | undefined;
+		warning: string | undefined;
+	}[];
+}
+
+interface Flag {
+	description: string;
+	flag: string;
+	type: string;
+}
+
+interface Option {
+	description: string;
+	docsSection:
+		| "core"
+		| "offline"
+		| "opt-out"
+		| "optional"
+		| "skip-disk"
+		| "skip-net";
+	multiple?: boolean;
+	type: string;
+}
+
+function logHelpTextSection(section: HelpTextSection): void {
+	console.log(" ");
+
+	console.log(`   ${chalk.black.bgGreenBright(section.sectionHeading)}`);
+
+	for (const subsection of section.subsections) {
+		if (subsection.warning) {
+			console.log(chalk.yellow(subsection.warning));
+		}
+
+		if (subsection.subheading) {
+			console.log(chalk.green(subsection.subheading));
+		}
+
+		for (const option of subsection.flags) {
+			const { description, flag, type } = option;
+			console.log(
+				chalk.cyan(
+					`
+      --${flag}${
+				type !== "boolean" ? ` (${chalk.cyanBright(type)})` : ""
+			}: ${description}`,
+				),
+			);
+		}
+	}
+}
+
+function createHelpTextSections(
+	options: Record<string, Option>,
+): HelpTextSection[] {
+	const core: HelpTextSection = {
+		sectionHeading: "Core options:",
+		subsections: [
+			{
+				flags: [],
+				subheading: undefined,
+				warning: undefined,
```

Comment:
[Cleanup] Just nitpicking: I'm not a huge fan of having `: undefined` properties when it'd be doable to make the property optional. Any reason not to make these `?: string` in `HelpTextSection`?

Actually, thinking further: `flags` could be made optional too... Just a few `?? []`s or similar instead in `help.ts`. Even less code!

Reply:
I've made `subheading` and `warning` optional. I can't see any reason why we would have a section without `flags` and we are pushing onto all of the empty arrays in the code currently, so I kind of prefer flags to be mandatory.